app/uk/gov/hmrc/gform/eval/ExpressionResult.scala

```diff
@@ -123,6 +134,7 @@ sealed trait ExpressionResult extends Product with Serializable {
     case t: DateResult    => false
     case t: PeriodResult  => er.ifPeriodResult(t.value.toTotalMonths == _.toTotalMonths)
     case t: AddressResult => false
+    case t: ListResult    => er.ifListResult(_ === t.list)
```

Comment:
There seems to be no `exists` semantics present here.
Shouldn't implementation be:
```
t.list.exists(_.identical(er))
```

Reply:
Not sure how useful this one is, I was just checking the list were having identical values i.e equality. Checking for contains when it says identical will be confusing may be?

Reply:
With this implementation only expressions of this form can ever be true:
```
${amount = amount}
```
This looks very limiting

More reasonable expressions which should potentially evaluate to true are of the form:
```
${amount = 1000}
```

Reply:
Perhaps best implementation is to support both cases:

```
er.ifListResult(_ === t.list) || t.list.exists(_ identical er)
```

Reply:
Ahh, now it makes sense. Done